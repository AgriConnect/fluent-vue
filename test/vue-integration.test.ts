import { createLocalVue, shallowMount } from '@vue/test-utils'

import { FluentBundle, ftl } from 'fluent'

// @ts-ignore
import App from './components/App.vue'

import FluentVue from '../src'

const localVue = createLocalVue()
localVue.use(FluentVue)

const bundle = new FluentBundle('en-US', {
  useIsolating: false
})

bundle.addMessages(ftl`
  message = Hello, { $name }!
`)

const fluent = new FluentVue({
  // locale: mainLocale,
  // fallbackLocale: getFallbackLocale(mainLocale),
  bundle
})

describe('vue integration', () => {
  it('translates messages', () => {
    // Act
    const mounted = shallowMount(App, {
      localVue,
      fluent
    } as any)

    // Assert
    expect(mounted.element).toMatchSnapshot()
  })

  it('clears instance on component destroy', () => {
    // Arrange
    const mounted = shallowMount(App, {
      localVue,
      fluent
    } as any)

    // Act
    mounted.destroy()

    // Assert
    expect(mounted.vm.$fluent).not.toBeUndefined()

    localVue.nextTick(() => {
      expect(mounted.vm.$fluent).toBeUndefined()
    })
  })
})
