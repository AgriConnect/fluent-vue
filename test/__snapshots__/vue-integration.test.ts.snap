// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`vue integration translates messages 1`] = `
<div>
  
  Hello, John!

</div>
`;
